bundles/org.openhab.binding.miio/src/main/java/org/openhab/binding/miio/internal/robot/RRMapDraw.java

```diff
@@ -0,0 +1,391 @@
+                g2d.setColor(COLOR_NO_GO_ZONES);
+                g2d.fill(noGo);
+                g2d.setColor(area == 9 ? Color.RED : Color.WHITE);
+                g2d.draw(noGo);
+            }
+        }
+    }
+
+    private void drawWalls(Graphics2D g2d, float scale) {
+        Stroke stroke = new BasicStroke(3 * scale);
+        g2d.setStroke(stroke);
+        for (float[] point : rmfp.getWalls()) {
+            float x = point[0] * scale;
+            float y = point[1] * scale;
+            float x1 = point[2] * scale;
+            float y1 = point[3] * scale;
+            g2d.setColor(Color.RED);
+            g2d.draw(new Line2D.Float(x, y, x1, y1));
+        }
+    }
+
+    private void drawRobo(Graphics2D g2d, float scale) {
+        float radius = 3 * scale;
+        Stroke stroke = new BasicStroke(2 * scale);
+        g2d.setStroke(stroke);
+        g2d.setColor(COLOR_CHARGER_HALO);
+        drawCircle(g2d, rmfp.getChargerX() * scale, rmfp.getChargerY() * scale, radius);
+        drawCenteredImg(g2d, scale / 8, "charger.png", rmfp.getChargerX() * scale, rmfp.getChargerY() * scale);
+        radius = 3 * scale;
+        g2d.setColor(COLOR_ROBO);
+        drawCircle(g2d, rmfp.getRoboX() * scale, rmfp.getRoboY() * scale, radius);
+        if (scale > 1.5) {
+            drawCenteredImg(g2d, scale / 15, "robo.png", rmfp.getRoboX() * scale, rmfp.getRoboY() * scale);
+        }
+    }
+
+    private void drawCircle(Graphics2D g2d, float x, float y, float radius) {
+        g2d.draw(new Ellipse2D.Double(x - radius, y - radius, 2.0 * radius, 2.0 * radius));
+    }
+
+    private void drawCenteredImg(Graphics2D g2d, float scale, String imgFile, float x, float y) {
+        URL image = getImageUrl(imgFile);
+        try {
+            if (image != null) {
+                BufferedImage addImg = ImageIO.read(image);
+                int xpos = Math.round(x - (addImg.getWidth() / 2 * scale));
+                int ypos = Math.round(y - (addImg.getHeight() / 2 * scale));
+                AffineTransform at = new AffineTransform();
+                at.scale(scale, scale);
+                AffineTransformOp scaleOp = new AffineTransformOp(at, AffineTransformOp.TYPE_BILINEAR);
+                g2d.drawImage(addImg, scaleOp, xpos, ypos);
+            } else {
+                logger.debug("Error loading image {}: File not be found.", imgFile);
+            }
+        } catch (IOException e) {
+            logger.debug("Error loading image {}: {}", image, e.getMessage());
+        }
+    }
+
+    private void drawGoTo(Graphics2D g2d, float scale) {
+        float x = rmfp.getGotoX() * scale;
+        float y = rmfp.getGotoY() * scale;
+        if (!(x == 0 && y == 0)) {
+            g2d.setStroke(new BasicStroke());
+            g2d.setColor(Color.YELLOW);
+            int x3[] = { (int) x, (int) (x - 2 * scale), (int) (x + 2 * scale) };
+            int y3[] = { (int) y, (int) (y - 5 * scale), (int) (y - 5 * scale) };
```

Comment:
is this correct?
```suggestion
            int y3[] = { (int) y, (int) (y - 5 * scale), (int) (y + 5 * scale) };
```

Reply:
It is suppose to draw a triangle. Bottom of the triangle at the xy